src/roles/vsd-deploy/tasks/renew_ejabberd_license.yml

```diff
@@ -0,0 +1,55 @@
+- block:
+
+  - name: Check that the ejabberdctl file exists
+    stat:
+      path: /opt/ejabberd/bin/ejabberdctl
+    register: ejabberdctl_result
+
+  - name: Install ejabberdctl if not present
+    shell: "/opt/vsd/install/install_ejabberd.sh -9 /opt/vsd/Packages -x {{ vsd_fqdn }}"
+    when: not ejabberdctl_result.stat.exists
+
+  - name: Find the ejabberd base directory
+    command: find /opt/ejabberd/lib/ -type d -name "ejabberd_nuage*"
+    register: ejabberd_base_dir
+
+  - name: Backup the current license files for ejabberd
+    command: tar zcf orig_backup.tgz ejabberd.beam ejabberd_admin.beam ejabberd_c2s.beam ejabberd_cluster.beam ejabberd_config.beam ejabberd_listener.beam ejabberd_sm.beam
+    args:
+      chdir: "{{ ejabberd_base_dir.stdout }}/ebin"
+
+  - name: Copy new ejabberd license file to root folder on VSD
+    copy:
+      dest: "/root/"
+      src: "{{ vsd_ejabberd_license_file }}"
+
+  - name: Untar the new ejabberd license file
+    command: "tar zxf /root/{{ vsd_ejabberd_license_file | basename }}"
+    args:
+      chdir: "{{ ejabberd_base_dir.stdout }}/ebin"
+
+  - name: Change ownership of licenses
+    command: "chown ejabberd:hadoopusers ejabberd_license.beam ejabberd_cluster.beam ejabberd_c2s.beam ejabberd.beam ejabberd_sm.beam ejabberd_router.beam ejabberd_listener.beam"
+    args:
+      chdir: "{{ ejabberd_base_dir.stdout }}/ebin"
+    when: not(ejabberd_base_dir.stdout is search("3.2.13_5"))
+
+  - name: Change ownership of licenses for version 3.2.13_5
+    command: "chown ejabberd:hadoopusers ejabberd_cluster.beam ejabberd_c2s.beam ejabberd.beam ejabberd_sm.beam ejabberd_router.beam ejabberd_listener.beam"
+    args:
+      chdir: "{{ ejabberd_base_dir }}/ebin"
+    when: ejabberd_base_dir.stdout is search("3.2.13_5")
+
+  - name: Start ejabberd
+    command: /opt/ejabberd/bin/ejabberdctl start
+    register: license_check_after_renew
+
+  - name: Check that new ejabberd license is valid
```

Comment:
What does the user see if this fails? Will they get an error that helps them to understand that there was a problem installing the license?

Reply:
If this fails, there is no meaningful error. It says could not connect ejabberd to vsd. But that error is not specific to license, it is the same error for many other conditions